core/node/fee_model/src/l1_gas_price/gas_adjuster/mod.rs

```diff
@@ -18,6 +18,97 @@ mod metrics;
 #[cfg(test)]
 mod tests;
 
+#[derive(Debug)]
+pub enum GasAdjusterClient {
+    L1(Box<DynClient<L1>>),
+    L2(Box<DynClient<L2>>),
+}
+
+struct GasAdjusterFee {
+    pub base_fee_per_gas: u64,
+    pub base_fee_per_blob_gas: U256,
+    pub l2_pubdata_price: U256,
+}
+
+impl GasAdjusterClient {
+    pub fn from_l1(inner: Box<DynClient<L1>>) -> Self {
```

Comment:
Nit: It would be more idiomatic to have `From<Box<DynClient<L1>>>` impl; ditto for `from_l2`.

Reply:
It unfortunately makes it harder in some places to use the method:

I have to do the following:

```
(Box::new(gateway.clone().into_client()) as Box::<DynClient::<L1>>).into()
```

which is more convoluted than just doing `from_l1`. I will keep both then